pkg/cmd/pr/pr.go

```diff
@@ -96,26 +103,26 @@ func NewCmdPullRequest() (*cobra.Command, *Options) {
 
 // Run implements the command
 func (o *Options) Run() error {
+	fmt.Println("Debug: Starting Run command...")
+
 	err := o.Validate()
 	if err != nil {
 		return fmt.Errorf("failed to validate: %w", err)
 	}
 
+	// Auto-discover git URL and commit details if not provided
 	if o.CommitMessage == "" || o.CommitTitle == "" || o.Application == "" {
-		// lets try discover the current git URL
-		if o.Application == "" || o.CommitMessage == "" {
-			gitURL, err := gitdiscovery.FindGitURLFromDir(o.Dir, true)
-			if err != nil {
-				log.Logger().Warnf("failed to find git URL %s", err.Error())
-			} else if gitURL != "" {
-				if o.Application == "" {
-					gitURLpart := strings.Split(gitURL, "/")
-					o.Application = gitURLpart[len(gitURLpart)-2] + "/" +
-						strings.TrimSuffix(gitURLpart[len(gitURLpart)-1], ".git")
-				}
-				if o.CommitMessage == "" {
-					o.CommitMessage = fmt.Sprintf("from: %s\n", gitURL)
-				}
+		gitURL, err := gitdiscovery.FindGitURLFromDir(o.Dir, true)
+		if err != nil {
+			fmt.Printf("Warning: Failed to discover git URL, error=%v\n", err)
+		} else if gitURL != "" {
+			if o.Application == "" {
+				gitURLParts := strings.Split(gitURL, "/")
+				o.Application = gitURLParts[len(gitURLParts)-2] + "/" +
+					strings.TrimSuffix(gitURLParts[len(gitURLParts)-1], ".git")
+			}
+			if o.CommitMessage == "" {
+				o.CommitMessage = fmt.Sprintf("from: %s\n", gitURL)
```

Comment:
Not a fan of the nested ifs and else ifs. I don't like the way it was originally done so would be good refactor the original implementation :)